src/lcals/FIRST_MIN-Cuda.cpp

```diff
@@ -80,24 +80,31 @@ void FIRST_MIN::runCudaVariantImpl(VariantID vid)
 
        FIRST_MIN_MINLOC_INIT;
 
+       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
        MyMinLoc* dminloc;
-       cudaErrchk( cudaMalloc( (void**)&dminloc, sizeof(MyMinLoc) ) );
-       cudaErrchk( cudaMemcpy( dminloc, &mymin, sizeof(MyMinLoc),
-                               cudaMemcpyHostToDevice ) );
+       MyMinLoc mymin_block[grid_size]; //per-block min value
```

Comment:
I'd be happier with an allocated buffer than a runtime sized array.

Reply:
In fact, would it be better if this was pinned memory in the first place so we could avoid the memcpy?

Reply:
@MrBurmark would that be a fair comparison with the RAJA variants?

Reply:
Right, I worry pinned memory here will make Cuda/HIP comparisons challenging.

Reply:
Pinned will be faster, so it will be a better baseline to compare against.